api/src/pcapi/core/offers/api.py

```diff
@@ -894,21 +894,18 @@ def add_criteria_to_offers(
 
 
 def reject_inappropriate_products(ean: str, send_booking_cancellation_emails: bool = True) -> bool:
-    products = models.Product.query.filter(models.Product.extraData["ean"].astext == ean).all()
+    product_query = models.Product.query.filter(models.Product.extraData["ean"].astext == ean)
+    product_ids = [p.id for p in product_query.with_entities(models.Product.id).all()]
 
-    if not products:
+    if not product_ids:
         return False
 
-    for product in products:
-        product.isGcuCompatible = False
-        db.session.add(product)
-
     offers_query = models.Offer.query.filter(
-        models.Offer.productId.in_(p.id for p in products),
+        models.Offer.productId.in_(product_query.with_entities(models.Product.id)),
```

Comment:
Je ne comprends pas bien cette modification : est-ce qu'il ne faudrait pas utiliser `product_ids` plutôt ?

Reply:
@Meewan me parle d'une limit dans le `in_`

Reply:
Le in est limité a 300 objets il est possible (mais peu probable) qu'on se retrouve a dépasser cette limite. donc une subquery est plus safe (bien qu'un peu moins performante je te l'accorde)